### ISSUE TITLE
Vague error stacks due to @babel/plugin-transform-react-jsx-source not being loaded

### ISSUE THREAD

**Issue by Symasia** [preactjs/preset-vite#62]
### Context:
Typical starter preact-ts template with Chakra UI installed. When using the `HStack`, `VStack` or `Stack` components with children nested in them, Vite throws the following error:

```
    Uncaught TypeError: can't define property "__": Object is not extensible
    Preact 78
    <anonymous> main.tsx:17
    Preact 78
    <anonymous> main.tsx:17
 ````

alongside a warning message saying that I should install the `@babel/plugin-transform-react-jsx-source` plugin:

```
Add @babel/plugin-transform-react-jsx-source to get a more detailed component stack. Note that you should not add it to production builds of your App for bundle size reasons.
```

The warning message is not printed to the console when say a syntax error is caused, and an appropriately helpful error message is thrown but with a unhelpful stack trace.

### Problem:

There are multiple aspects to look at:

1) If the plugin is required to provide better stack traces, why is it not loaded by default? If I'm understanding right the preact plugin itself is more or less a copy of `plugin-react` but based on a [comment](https://github.com/vitejs/vite/blob/7c4accb8dec3204948d7b73e5386e789661f9198/packages/plugin-react/src/index.ts#L267) in the said plugin, it's only required for versions of React before v17. 
If it's only required for classic runtimes then why the warning message?

2) It seems that loading the plugin via `babel.plugins` and `babel.babelrc` props don't make the warning message go away. This could mean that the plugin is not loaded or at least in the correct order(?).

3) Putting aside the lack of a helpful error message and a stack trace, why is this particular TypeError occurring? This is code that runs perfectly fine with the `react-ts` template.

 ### What I've tried:

**Here is a reproducible demo: https://stackblitz.com/edit/vitejs-vite-pzmvj6**

I have tested with both `preact` and `preact-ts` both yield the same result. Both of them show the same error message and warning log.

However, when testing with the **[`react-ts` template](https://stackblitz.com/edit/vitejs-vite-3e92e8)**, neither the error nor the warning message is printed to the console.

I have tried using the `vite-plugin-babel` plugin to try and load the plugin but to no avail. 

If you build the project as is with the error message and run it, you'll notice that any children nested inside of `HStack`, `VStack`, or `Stack` do not render. The code exists in the bundle, however.

---

Not too sure about the title of this issue, change it to something appropriate if need be. 

For the moment, I have switched to using React but it would be nice to see this fixed. 

I'm willing to help in any way I can.

**Comment by marvinhagemeister** [comment]
@Symasia Found the root cause. That was an odd one to figure out. Basically, we show that warning when the development JSX transform isn't active. But in our it _was_ active, but the problem was that we received an object that just looked close enough to our own, so it was treated as being a valid virtual dom node. If that wouldn't have happened, we'd thrown way earlier.

Took a while to figure out what was going on, but the culprit is the `@chakra-ui/react-children-utils` package, which is ultimately included when importing `@chakra-ui/react`. They forgot to declare `react` as a peer dependency, which lead to their build tool inlining a full copy of React when publishing to npm. That's a bug in chakra ui and I filed a PR to address that on their end: https://github.com/chakra-ui/chakra-ui/pull/6789

When Preact attempted to render `HStack` we got a virtual dom node created by react instead of our compatibility layer. And because react's virtual dom node looks very similar to the one generated by `preact/compat`, we didn't pick that up.

I'm afraid the only solution is to wait until they merge the fix and publish a new version of `@chakra-ui/react-children-utils`.
